commands/removerole.js

```diff
@@ -0,0 +1,142 @@
+const Discord = require("discord.js")
+const hutil = require("../utilities/hutil.js")
+const reaction_message_model = require("../models/reaction_roles_message.js")
+const mongoose = require('mongoose');
+mongoose.connect('mongodb+srv://XtratoS:UfTMXyIAxTN6Xr3q@nbot-dg4ad.mongodb.net/rbot?retryWrites=true', { useNewUrlParser: true })
+
+/**
+ * Has 2 usages:
+ * -If a role/emoji is provided, removed that role/emoji from the specified message,
+ * -If no role/emoji provdided, removes all the roles/emojis from the specified message.
+ */
+module.exports = async (bot, message, args) => {
+    if (!args[0] || args[0] == "help") {
+        let help = new Discord.RichEmbed()
+            .setColor("#FFFF00")
+            .addField("Usage", "removerole <message_id> <role>/<emoji>")
+            .addField("More info", "*Has 2 usages*:\nIf a role/emoji is provided, removed that role/emoji from the specified message,\n-If no role/emoji provdided, removes all the roles/emojis from the specified message.")
+        return message.channel.send(help).then(r => r.delete(30000))
+    }
+    if(!message.member.hasPermission("MANAGE_ROLES")){
+        message.reply("You don't have permission").then(async m=>{
+            await message.delete()
+            return m.delete(15000)
+        })
+    }
+    var message_id = args[0]
+    if (!args[1]) {
+        // remove the whole record that includes this message
+        let removeall = new Discord.RichEmbed()
+            .setColor("#FFA500")
+            .addField("Warning", "You didn't provide a role/emoji to remove, would you like to remove all the roles/emojis bounded to this message?")
+            .addField("Respone", "Type **Yes** to overwrite or anything else to Cancel")
+            .setDescription("*This command will be automatically cancelled in 60 seconds*")
+        msg = message.channel.send(removeall)
+        const filter = m => m.author.id === message.author.id
+        var response = await message.channel.awaitMessages(filter, { max: 1, time: 60000 }).then(async collected => {
+            if (!collected.first()) {
+                msg.edit(new Discord.RichEmbed()
+                    .setColor("#FF0000")
+                    .addField("Failture", "Command timed out")
+                ).then(m => m.delete(30000))
+                return false
+            }
+            if (collected.first().content.toLowerCase() == "yes") {
+                return true
+            } else {
+                msg.edit(new Discord.RichEmbed()
+                    .setColor("#FFFF00")
+                    .addField("Information", "Command Cancelled")
+                ).then(m => m.delete(30000))
+                return false
+            }
+        }).catch(r => { console.log(r) })
+        if (!response) {
+            return
+        }
+        reaction_message_model.deleteOne({
+            messageID: message_id
+        }, (err, rem) => {
```

Comment:
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion
        }, function(err, rem) {
```